### DISCUSSION TITLE
QiitaAPIのアクセストークンについて

### DISCUSSION THREAD

**Discussion by HaruhikoMotokawa** [increments/qiita-discussions#883] [要望・アイデア・不具合]
いつもお世話になっています。
こちらで投稿先があっているか判断しかねますが、投稿させていただきます。
なお、私はiOSアプリ開発の学習１年、Flutterの実務約半年ほどの経歴です。
間違いがございましたら、ご指導いただければ幸いです。

## 【質問】 QiitaAPIのアクセストークンの有効期限の明記がない
こちらの[ドキュメント](https://qiita.com/api/v2/docs#アクセストークン-1)を読む限り、有効期限の明記がありません。
アクセストークンは通常、数十分から数時間程度の有効期限を設けるものと認識しています。
QiitaAPIのアクセストークンには有効期限が設けられていないのでしょうか？

## 【提案】 QiitaAPIのリフレッシュトークンの実装

質問の回答を仮に「有効期限がない」とするならば、リフレッシュトークンの導入と、アクセストークンの有効期限の設定することを要望したいと思います。

近年のログインまわりの仕様では、アクセストークンとリフレッシュトークンをセットで導入していることが多いのではないでしょうか？
私は実務経験が浅いので、あくまで私の現在の現場であったり、読んだ記事などの観測範囲でお話しています。
その上でこの二つをセットで導入することがスタンダードなのであれば、技術に関する記事を扱っているサービスである「Qiita」であるからこそ、そのスタンダードな仕様を踏襲するべきでは？と考えた次第です。

確かにアクセストークンだけで気軽にログインまわりを実装できるので、初心者には入りやすいです。
私もそうでした。
しかし、実務に入ってリフレッシュトークンというものがあり、それがスタンダードと言われた際には戸惑いました。
導入に際してのコストもあるし、サービスで扱う内容次第ではリフレッシュトークンを使わない事例があることも承知しています。

この点に関して、どのようにお考えであるかを教えていただければ幸いです。

**Comment by P-SiZK** [comment]
@HaruhikoMotokawa 
お問合せいただきありがとうございます。

ご指摘の通り、アクセストークンには有効期限が設けられていません。
また、リフレッシュトークンに関してですが、新しいバージョンのAPI設計を行う際に検討したいと考えています。

**Reply by HaruhikoMotokawa** [reply]
@P-SiZK 
ご丁寧にご返信いただきありがとうございます。
お忙しいとは思いますが、ご検討よろしくお願いいたします🙇‍♂️
